packages/client/src/modules/pastry/PastryForm.js

```diff
@@ -0,0 +1,159 @@
+import { useMutation } from "@apollo/client";
+import { Button, Grid, TextField, Typography } from "@mui/material";
+import { useFormik } from "formik";
+import React from "react";
+import * as yup from "yup";
+import { SelectCategory } from "../category/SelectCategory";
+import { ADD_PASTRY, UPDATE_PASTRY } from "./mutations";
+
+const fieldStyle = { width: 500 };
+
+const validationSchema = yup.object({
+  name: yup.string().required().label("Name"),
+  origin: yup.string().required().label("Origin"),
+  description: yup.string().required().label("Description"),
+  price: yup.number().required().label("Price").min(0.75),
+  categoryId: yup.string().required().label("Category"),
+});
+
+export const PastryForm = ({ id, initialValues, onClose }) => {
+  const mutation = id !== undefined ? UPDATE_PASTRY : ADD_PASTRY;
+
+  const [savePastry, { loading, error }] = useMutation(mutation, {
+    refetchQueries: ["GET_PASTRIES", "GET_PASTRY"],
+    onCompleted: () => {
+      if (onClose !== undefined) onClose();
+    },
+  });
+
+  const {
+    values,
+    errors,
+    handleChange,
+    handleBlur,
+    handleSubmit,
+    handleReset,
+  } = useFormik({
+    initialValues,
+    validationSchema,
+    onSubmit: async (values) => {
+      console.log(`Pastry ID: ${id}`);
+      console.log("Values:", values);
+
+      const { title, origin, description, imgsrc, categoryId, price } = values;
+      const input = { title, origin, description, imgsrc, categoryId, price };
```

Comment:
The server was expecting 'name' instead of 'title'. Change that and apply the categoryId fix and this will work.